### PR TITLE
Fix compilation error on big-endian platforms

### PR DESCRIPTION
**Pull request by ltrk2**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

File: src/Functions/FunctionsCodingIP.cpp [hunk 1]
```diff
@@ -580,7 +580,7 @@ class FunctionIPv4ToIPv6 : public IFunction
 #if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
             unalignedStoreLittleEndian<UInt64>(buf + 8, 0x00000000FFFF0000ull | (static_cast<UInt64>(ntohl(in)) << 32));
 #else
-            unalignedStoreLittleEndian<UInt64>(buf + 8, 0x00000000FFFF0000ull | (static_cast<UInt64>(__builtin_bswap32(in))) << 32));
```
Comment (vitlibar) [inline src/Functions/FunctionsCodingIP.cpp:583]: What was wrong with `__builtin_bswap32`?

Comment (vitlibar) [inline src/Functions/FunctionsCodingIP.cpp:583]: A superfluous parenthesis?

**Review by vitlibar — APPROVED** [review]

**Comment by robot-ch-test-poll1** [comment]
This is an automated comment for commit d1cb332fc6d62b5ea262cc84fb0fee451d2c8eee with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/50406/d1cb332fc6d62b5ea262cc84fb0fee451d2c8eee/ci_running.html)
The overall status of the commit is 🟢 success

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>Style Check</td><td>Runs a set of checks to keep the code style clean. If some of tests failed, see the related log from the report</td><td>🟢 success</td></tr>
<tr><td>security/snyk (ClickHouse)</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>🟢 success</td></tr>
</table>

File: src/Functions/FunctionsCodingIP.cpp (hunk 1 above)
Comment (ltrk2) [reply src/Functions/FunctionsCodingIP.cpp:583]: Yes, exactly.

Comment (ltrk2) [reply src/Functions/FunctionsCodingIP.cpp:583]: There wasn't anything wrong it, it's just `std::byteswap` is a more generic alternative starting from C++23.

**Comment by ltrk2** [comment]
@vitlibar could you please merge this in case the tests look okay?

**Pull request merged**